### PR TITLE
feat: booking confirmation header card 

### PR DESCRIPTION
**Pull request by fandihungryhub**
https://3.basecamp.com/5190892/buckets/31862334/card_tables/cards/5985638915
https://3.basecamp.com/5190892/buckets/31862334/card_tables/cards/5985646500
all package type (ayce, pp, hah, xp)

### CODE REVIEW

File: components/booking/Confirmation/BookingCardImage.vue [hunk 1]
```diff
@@ -0,0 +1,110 @@
+<template>
+  <div class="relative w-fit rounded-xl" :class="packageType.boxShadow">
+    <div v-if="!showQrCode" class="absolute p-2.5" @click="onShowQrCode">
+      <HhImage 
+        alt="QR Code" 
+        src="/image/sample-qrcode-small.png"
```
Comment (afif-hh) [inline components/booking/Confirmation/BookingCardImage.vue:6]: nama folder `component` nya c kecil, image qr nya dari props

File: components/booking/Confirmation/BookingCardImage.vue [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+        <div>Booking ID: <span class="text-red-dark">{{ props.bookingId }}</span></div>
+        <div><HhIcon /></div>
+      </div>
+    </div>
+  </div>
+</template>
+
+<script setup lang="ts">
+import { ref, PropType, computed } from "vue";
+import HhImage from '@/components/HhImage.vue';
```
Comment (afif-hh) [inline components/booking/Confirmation/BookingCardImage.vue:47]: @ nya diganti ke ~ aja

File: components/booking/Confirmation/BookingCardImage.vue [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+            :src="props.qrCodeUrl"
+            width="182"
+            height="182"
+            class="rounded-xl h-[182px]"
+          />
+        </div>
+      </div>
+      <div class="flex items-center justify-between bg-white rounded-b-xl text-xs font-bold p-2.5">
+        <div>Booking ID: <span class="text-red-dark">{{ props.bookingId }}</span></div>
+        <div><HhIcon /></div>
```
Comment (afif-hh) [inline components/booking/Confirmation/BookingCardImage.vue:39]: hhIcon buat apa mas ?

File: components/booking/Confirmation/BookingCardImage.vue [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+  }
+});
+
+const showQrCode = ref(false);
+
+const packageType = computed(() => {
+  if (props.packageType === "ayce") {
+    return { text: "ALL YOU<br>CAN EAT", bgColor: "bg-red-dark", boxShadow: "box-shadow-ayce", };
+  } else if (props.packageType === "pp") {
+    return { text: "PARTY<br>PACK", bgColor: "bg-mint-600", boxShadow: "box-shadow-pp", };
```
Comment (afif-hh) [inline components/booking/Confirmation/BookingCardImage.vue:79]: warna tipe package ada di tailwindconfig mas, bisa di cek

File: components/booking/Confirmation/BookingCardImage.vue [hunk 5]
```diff
@@ -0,0 +1,110 @@
...
+<style>
+.box-shadow-ayce {
+  box-shadow: 0px 25px 45px rgba(223, 37, 42, 0.4);
+}
+
+.box-shadow-pp {
+  box-shadow: 0px 25px 45px rgba(188, 100, 255, 0.4);
+}
+
+.box-shadow-hah {
```
Comment (afif-hh) [inline components/booking/Confirmation/BookingCardImage.vue:103]: box shadown nya pakai warna di tailwindconfig mas

File: components/booking/Confirmation/BookingHeader.vue [hunk 6]
```diff
@@ -0,0 +1,137 @@
...
+    <div class="main-card w-full flex justify-center mb-3">
+      <BookingCardImage 
+        :booking-id="props.bookingId" 
+        :image-cover="props.imageCover"
+        :qr-code-url="props.qrCodeUrl"
+        :package-type="props.packageType"
+      />
+    </div>
+    <div class="info mb-3 mx-4">
+      <div class="mb-1.5 text-base text-white font-bold">Copper Buffet</div>
```
Comment (afif-hh) [inline components/booking/Confirmation/BookingHeader.vue:16]: copper buffet masih hard code

File: components/booking/Confirmation/BookingHeader.vue [hunk 7]
```diff
@@ -0,0 +1,137 @@
...
+        <span class="text-sm mr-3">{{ props.kid }}</span>
+      </div>
+    </div>
+  </div>
+</template>
+
+<script setup lang="ts">
+import { ref, PropType, computed } from "vue";
+import dayjs from "~/lib/dayjs";
+import BookingCardImage from "@/components/booking/Confirmation/BookingCardImage.vue";
```
Comment (afif-hh) [inline components/booking/Confirmation/BookingHeader.vue:34]: @ diganti ~

File: components/booking/Confirmation/BookingHeader.vue [hunk 8]
```diff
@@ -0,0 +1,137 @@
...
+  imageCover: {
+    type: String,
+    default: null,
+  },
+  qrCodeUrl: {
+    type: String,
+    default: null,
+  },
+  packageType: {
+    type: String as PropType<"ayce" | "pp" | "hah" | "xp">,
```
Comment (afif-hh) [inline components/booking/Confirmation/BookingHeader.vue:60]: proptype ayce dll bisa import dari SupportedPackageType di types/Pack

File: components/booking/Confirmation/BookingHeader.vue [hunk 9]
```diff
@@ -0,0 +1,137 @@
...
+    default: "ayce",
+  },
+  dateTime: {
+    type: String,
+    default: null,
```
Comment (afif-hh) [inline components/booking/Confirmation/BookingHeader.vue:65]: props yg penting di jadiin required aja mas, kyk adult, date time, package type

File: components/booking/Confirmation/BookingHeader.vue [hunk 10]
```diff
@@ -0,0 +1,137 @@
...
+  return null;
+});
+
+function onShowQrCode() {
+  showQrCode.value = true;
+}
+</script>
+
+<style>
+.bg-gradient-ayce {
```
Comment (afif-hh) [inline components/booking/Confirmation/BookingHeader.vue:122]: dicoba pakai warna dari tailwindconfig, biar kalau ada perubahan code warna terpusat

File: components/icons/HhIcon.vue [hunk 11]
```diff
@@ -0,0 +1,14 @@
...
+  <path d="M17.338 4.41491C17.3198 4.52364 17.2514 4.62374 17.1367 4.67204L9.2621 8.04231C9.09403 8.11301 8.90773 8.11301 8.74123 8.04231L0.86195 4.67041C0.750575 4.62211 0.683975 4.52714 0.66395 4.42354C0.63245 4.27164 0.698825 4.10084 0.86195 4.03014L8.74145 0.658005C8.90795 0.587305 9.09425 0.587305 9.26232 0.658005L17.1369 4.02827C17.2982 4.09734 17.3648 4.26464 17.3383 4.41491H17.338Z" fill="#DF252A"/>
+  <path d="M15.1621 6.49074V12.5791C15.1621 12.7016 15.0906 12.8138 14.9808 12.8605L14.0538 13.2574C13.8607 13.3402 13.6494 13.1935 13.6494 12.9779V10.6464L10.0866 12.1719V14.7519C10.0866 14.876 10.015 14.9866 9.9052 15.0331L9.1132 15.373C9.04165 15.4024 8.9602 15.4024 8.88843 15.373L8.09643 15.0331C7.98663 14.9864 7.91508 14.876 7.91508 14.7519V12.1719L4.35063 10.6464V12.9779C4.35063 13.1935 4.13935 13.3402 3.94788 13.2574L3.0193 12.8605C2.9095 12.8138 2.83795 12.7016 2.83795 12.5791V6.49074C2.83795 6.48561 2.83795 6.48048 2.83953 6.47534L4.3504 7.12238V9.69721L7.91485 11.2227V8.64791L8.73363 8.99814C8.90328 9.07071 9.09633 9.07071 9.26778 8.99814L10.0866 8.64791V11.2227L13.6494 9.69721V7.12238L15.1603 6.47534C15.1621 6.48048 15.1621 6.48561 15.1621 6.49074Z" fill="black"/>
```
Comment (afif-hh) [inline components/icons/HhIcon.vue:6]: hh icon buat apa mas

**Review by afif-hh — CHANGES_REQUESTED** [review]

File: components/booking/Confirmation/BookingCardImage.vue (hunk 1 above)
Comment (fandihungryhub) [reply components/booking/Confirmation/BookingCardImage.vue]: updated

File: components/booking/Confirmation/BookingCardImage.vue (hunk 2 above)
Comment (fandihungryhub) [reply components/booking/Confirmation/BookingCardImage.vue]: updated

File: components/booking/Confirmation/BookingCardImage.vue (hunk 3 above)
Comment (fandihungryhub) [reply components/booking/Confirmation/BookingCardImage.vue:40]: Screenshot 2023-04-05 at 10 10 33

File: components/booking/Confirmation/BookingCardImage.vue (hunk 4 above)
Comment (fandihungryhub) [reply components/booking/Confirmation/BookingCardImage.vue:81]: gak sama mas

File: components/booking/Confirmation/BookingCardImage.vue (hunk 5 above)
Comment (fandihungryhub) [reply components/booking/Confirmation/BookingCardImage.vue:105]: gak ada mas

File: components/booking/Confirmation/BookingHeader.vue (hunk 6 above)
Comment (fandihungryhub) [reply components/booking/Confirmation/BookingHeader.vue]: updated

File: components/booking/Confirmation/BookingHeader.vue (hunk 7 above)
Comment (fandihungryhub) [reply components/booking/Confirmation/BookingHeader.vue]: updated

File: components/booking/Confirmation/BookingHeader.vue (hunk 8 above)
Comment (fandihungryhub) [reply components/booking/Confirmation/BookingHeader.vue]: updated

File: components/booking/Confirmation/BookingHeader.vue (hunk 9 above)
Comment (fandihungryhub) [reply components/booking/Confirmation/BookingHeader.vue]: updated

File: components/booking/Confirmation/BookingHeader.vue (hunk 10 above)
Comment (fandihungryhub) [reply components/booking/Confirmation/BookingHeader.vue:127]: bikin linear-gradient nya susah mas

File: components/icons/HhIcon.vue (hunk 11 above)
Comment (fandihungryhub) [reply components/icons/HhIcon.vue:6]: Screenshot 2023-04-05 at 10 10 33

**Pull request merged**